x-pack/plugins/apm/public/components/app/service_details/service_icons/index.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiLoadingSpinner } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { useUrlParams } from '../../../../context/url_params_context/use_url_params';
+import { FETCH_STATUS, useFetcher } from '../../../../hooks/use_fetcher';
+import { getAgentIcon } from '../../../shared/AgentIcon/get_agent_icon';
+import { CloudDetails } from './cloud_details';
+import { ContainerDetails } from './container_details';
+import { IconPopover } from './icon_popover';
+import { ServiceDetails } from './service_details';
+
+interface Props {
+  serviceName: string;
+}
+
+const cloudIcons: Record<string, string> = {
+  gcp: 'logoGCP',
+  aws: 'logoAWS',
+  azure: 'logoAzure',
+};
+
+function getCloudIcon(provider?: string) {
+  if (provider) {
+    return cloudIcons[provider];
+  }
+}
+
+export function ServiceIcons({ serviceName }: Props) {
+  const {
+    urlParams: { start, end },
+    uiFilters,
+  } = useUrlParams();
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
```

Comment:
It looks like you store the same state twice in `ServiceIcons` and in `IconPopover`

Reply:
@sqren  This is necessary because I need three different states one for each icon and one state that controls when the details API should be called.